InShine-Studios/Horror-3D
# [rework] Inventory: Make Item class NOT a child of interactable class

### Main changes
Interactable.cs:
- AudioPlayerObj dijadiin private
- PlayAudio(string name) dijadiin private

Item.cs:
- Tambahin AudioPlayerObj dan PlayAudio (sama kyk punya Interactable.cs), dibikin private juga
- Rename OnInteraction() jadi Pick() (pake ctrl+r+r, biarin aja error)
- Apus fungsi OnInteraction() yg barusan direname jadi Pick(), Pick() yg lama biarin aja
- Set Pick yg lama jadi public
- Apus interface IItem dan bikin interface baru (autogenerate aja)

Detectors:
- Pindahin semua fungsi dan variable di ObjectDetector.cs ke InteractableDetector.cs dan ItemDetector.cs terus sesuain tipenya (InteractableDetector buat Interactable aja, ItemDetector buat Item aja)
- Delete class ObjectDetector.cs
- Ilangin inheritance InteractableDetector.cs dan ItemDetector.cs ke ObjectDetector.cs
- GhostSimulationInteractableDetector.cs inherits ItemDetector.cs

How to check:
- Check the codes
- Run tests

### Tasks List
- [close #342 ]
